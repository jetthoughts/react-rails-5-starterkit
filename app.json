{
  "name": "react-rails-5-starterkit",
  "env": {
    "RAILS_ENV": "staging",
    "RACK_ENV": "staging",
    "NODE_ENV": "production",
    "BUNDLE_WITHOUT": "production:development:test",
    "DB_SEED_ON_REDEPLOY": {
      "description": "Reset database on redeploy",
      "value": "true"
    },
    "RAILS_MAX_THREADS": "5",
    "WEB_CONCURRENCY": "2",
    "HEROKU_APP_NAME": {
      "required": true
    },
    "RAILS_LOG_TO_STDOUT": {
      "required": true
    },
    "RAILS_SERVE_STATIC_FILES": {
      "required": true
    }
  },
  "addons": [
    "heroku-postgresql:hobby-dev",
    "memcachier:dev"
  ],
  "buildpacks": [
    {
      "url": "heroku/ruby"
    }
  ]
}
