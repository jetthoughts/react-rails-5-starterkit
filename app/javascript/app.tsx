import * as React from "react"

interface AppProps {
  name: string
}

const App: React.SFC<AppProps> = (props) => (
  <div>Hello {props.name}!</div>
)

App.defaultProps = {
  name: 'David',
}

export default App
