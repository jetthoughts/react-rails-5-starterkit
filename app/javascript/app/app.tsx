import * as React from 'react'
import './app.scss'

interface AppProps {
  name: string
}

const App: React.SFC<AppProps> = (props) => (
  <p>Hello {props.name}!</p>
)

App.defaultProps = {
  name: 'David',
}

export default App
