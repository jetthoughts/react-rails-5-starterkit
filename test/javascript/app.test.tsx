import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { default as App } from 'Javascript/app'

it('renders app component with a custom name', () => {
    const component = renderer.create(<App name={'Dmitriy'} />)
    expect(component).toMatchSnapshot()
})

it('renders app component with a default name', () => {
    const component = renderer.create(<App />)
    expect(component).toMatchSnapshot()
})
