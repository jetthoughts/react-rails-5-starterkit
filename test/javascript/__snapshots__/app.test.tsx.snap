// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component with a custom name 1`] = `
<div>
  Hello 
  Dmitriy
  !
</div>
`;

exports[`renders app component with a default name 1`] = `
<div>
  Hello 
  David
  !
</div>
`;
